### PR TITLE
edit logistic regression notebook to add sparse vectors

### PR DESCRIPTION
**Pull request by lijinf2**

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: notebooks/logistic-regression.ipynb [hunk 1]
```diff
@@ -647,6 +647,108 @@
...
+   "source": [
+    "def sparse_vectors_compat(EstimatorClass):\n",
+    "    from pyspark.ml.evaluation import MulticlassClassificationEvaluator\n",
+    "    from pyspark.ml.feature import CountVectorizer, RegexTokenizer\n",
+    "    from pyspark.sql import Row\n",
+    "    from sklearn.datasets import fetch_20newsgroups\n",
+    "\n",
+    "    # prepare dataframe\n",
+    "    try:\n",
+    "        twenty_train = fetch_20newsgroups(subset=\"train\", shuffle=True, random_state=42)\n",
```
Comment (eordentlich) [inline notebooks/logistic-regression.ipynb:672]: Does this take a while?   Done twice currently.

File: notebooks/logistic-regression.ipynb [hunk 2]
```diff
@@ -647,6 +647,108 @@
...
+    "    df = spark.createDataFrame(data)\n",
+    "    \n",
+    "    # convert text to sparse vector\n",
+    "    tokenizer = RegexTokenizer(inputCol=\"text\", outputCol=\"tokens\")\n",
+    "    df = tokenizer.transform(df)\n",
+    "    cv = CountVectorizer(inputCol=\"tokens\", outputCol=\"features\")\n",
+    "    cv_model = cv.fit(df)\n",
+    "    df = cv_model.transform(df)\n",
+    "\n",
+    "    df_train, df_test = df.randomSplit([0.8, 0.2])\n",
```
Comment (eordentlich) [inline notebooks/logistic-regression.ipynb:696]: Better to have same data/test/train split for gpu and cpu.   This will be different for each call.

File: notebooks/logistic-regression.ipynb [hunk 3]
```diff
@@ -647,6 +647,108 @@
...
+    "\n",
+    "    lr = EstimatorClass(\n",
+    "        regParam=0.01,\n",
+    "        maxIter=100,\n",
+    "        fitIntercept=True,\n",
+    "        standardization=False,\n",
```
Comment (eordentlich) [inline notebooks/logistic-regression.ipynb:702]: Add a markdown cell about standardization needing to be false to keep data sparse, for now, will be fixed in 24.02.

**Review by eordentlich — APPROVED** [review]

File: notebooks/logistic-regression.ipynb (hunk 3 above)
Comment (lijinf2) [reply notebooks/logistic-regression.ipynb:718]: Added the cell under "Spark Vectors" title to explain this point.

File: notebooks/logistic-regression.ipynb (hunk 1 above)
Comment (lijinf2) [reply notebooks/logistic-regression.ipynb]: Revised to move the dataframe preparation code out so that it is executed once.

File: notebooks/logistic-regression.ipynb (hunk 2 above)
Comment (lijinf2) [reply notebooks/logistic-regression.ipynb]: Added seed=0.

**Comment by lijinf2** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍
